src/module.c

```diff
@@ -785,8 +790,8 @@ int TSDB_madd(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
     RedisModuleString *curTimeStr = NULL;
 
     RedisModule_ReplyWithArray(ctx, (argc - 1) / 3);
-    RedisModuleString **replication_data = malloc(sizeof(RedisModuleString *) * (argc - 1));
-    size_t replication_count = 0;
+    RedisModuleString **replArgv = malloc((argc - 1) * sizeof *replArgv);
```

Comment:
Since this is a temp local buffer probably `alloca()` would be better here (also below).